mx.sulong/tools.py

```diff
@@ -0,0 +1,176 @@
+import mx
+import os
+
+import mx_sulong
+from shutil import copyfile
+
+class ProgrammingLanguage(object):
+    class PL(object):
+        def __init__(self, name, exts):
+            self.name = name
+            self.exts = exts
+
+    exts = {}
+
+    @staticmethod
+    def register(name, *exts):
+        lang = ProgrammingLanguage.PL(name, exts)
+        setattr(ProgrammingLanguage, name, lang)
+        for ext in exts:
+            ProgrammingLanguage.exts[ext] = lang
+
+    @staticmethod
+    def lookup(extension):
+        return ProgrammingLanguage.exts.get(extension, None)
+
+ProgrammingLanguage.register('FORTRAN', 'f90', 'f', 'f03')
+ProgrammingLanguage.register('C', 'c')
+ProgrammingLanguage.register('C_PLUS_PLUS', 'cpp', 'cc', 'C')
+ProgrammingLanguage.register('OBJECTIVE_C', 'm')
+ProgrammingLanguage.register('LLVMIR', 'll')
+ProgrammingLanguage.register('LLVMBC', 'bc')
+ProgrammingLanguage.register('LLVMSU', 'su')
+ProgrammingLanguage.register('EXEC', 'out')
+
+
+class Optimization(object):
+    class Opt(object):
+        def __init__(self, name, flags):
+            self.name = name
+            self.flags = flags
+
+    @staticmethod
+    def register(name, *flags):
+        setattr(Optimization, name, Optimization.Opt(name, list(flags)))
+
+Optimization.register('NONE')
+Optimization.register('BB_VECTORIZE')
+Optimization.register('O1', '-O1')
+Optimization.register('O2', '-O2')
+Optimization.register('O3', '-O3')
+
+
+class Tool(object):
+    def supports(self, language):
+        return language in self.supportedLanguages
+
+class ClangCompiler(Tool):
+    def __init__(self):
+        self.name = 'clang'
+        self.supportedLanguages = [ProgrammingLanguage.C, ProgrammingLanguage.C_PLUS_PLUS, ProgrammingLanguage.OBJECTIVE_C]
+
+    def run(self, inputFile, outputFile, flags):
+        try:
+            f = open('/dev/null', 'w')
+            return mx.run([mx_sulong.findLLVMProgram('clang'), '-S', '-emit-llvm', '-o', outputFile] + flags + [inputFile], out=f, err=f)
+        except SystemExit:
+            print 'Cannot compile %s' % (inputFile)
+        return -1
+
+    def compileReferenceFile(self, inputFile, outputFile, flags):
+        try:
+            f = open('/dev/null', 'w')
```

Comment:
That only works for Unix system. Could you solve that in a platform-independent way?